mlflow/utils/databricks_utils.py

```diff
@@ -497,37 +532,66 @@ def get_databricks_host_creds(server_uri=None):
-        dbutils = _get_dbutils()
-        if dbutils:
-            # Prefix differentiates users and is provided as path information in the URI
-            key_prefix = path
-            host = dbutils.secrets.get(scope=profile, key=key_prefix + "-host")
-            token = dbutils.secrets.get(scope=profile, key=key_prefix + "-token")
-            if host and token:
-                config = DatabricksConfig.from_token(host=host, token=token, insecure=False)
-    if not config or not config.host:
-        _fail_malformed_databricks_auth(profile)
 
-    insecure = hasattr(config, "insecure") and config.insecure
+    from databricks.sdk import WorkspaceClient
 
-    if config.username is not None and config.password is not None:
-        return MlflowHostCreds(
-            config.host,
-            username=config.username,
-            password=config.password,
-            ignore_tls_verification=insecure,
-        )
-    elif config.token:
-        return MlflowHostCreds(config.host, token=config.token, ignore_tls_verification=insecure)
-    _fail_malformed_databricks_auth(profile)
+    profile, _ = get_db_info_from_uri(server_uri)
+    try:
+        # Using databricks-sdk to create Databricks WorkspaceClient instance,
+        # If authentication is failed, exception is raised.
+        # databricks-sdk supports many kinds of authentication ways,
+        # it will try to read authentication information by the following ways:
+        # 1. Read dynamic generated token via databricks `dbutils`.
+        # 2. parse relevant environment variables (such as DATABRICKS_HOST + DATABRICKS_TOKEN
+        #    or DATABRICKS_HOST + DATABRICKS_CLIENT_ID + DATABRICKS_CLIENT_SECRET)
+        #    to get authentication information
+        # 3. parse ~/.databrickscfg file (generated by databricks-CLI command-line tool)
+        #    to get authentication information.
+        # databricks-sdk is designed to hide authentication details and
+        # support various authentication ways, so that it does not provide API
+        # to get credential values. Instead, we can use ``WorkspaceClient``
+        # API to invoke databricks shard restful APIs.
+        ws_client = WorkspaceClient(profile=profile)
+        ws_client.config.retry_timeout_seconds = \
+            MLFLOW_DATABRICKS_ENDPOINT_HTTP_RETRY_TIMEOUT.get()
+        host = ws_client.config.host
+        return MlflowHostCreds(host=host, databricks_workspace_client=ws_client)
+    except Exception as e:
+        _logger.info(f"Creating databricks SDK workspace client failed, error: {repr(e)}")
+        # There are a few other ways to read credentials that databricks-sdk hasn't supported yet,
+        # we handle them here as the fallback approaches.
+        config = None
+
+        for provider in [
+            TrackingURIConfigProvider(server_uri),
+            SparkTaskContextConfigProvider(),
+            DatabricksModelServingConfigProvider(),
+        ]:
+            if provider:
```

Comment:
looks like `provider` is always true.

Reply:
`_dynamic_token_config_provider` is None if it is not in databricks runtime

Reply:
I don't see `_dynamic_token_config_provider` here.